### ISSUE TITLE
cilium not work with default metrics-server release name install in kube-system namespace

### ISSUE THREAD

**Issue by qianlongzt** [cilium/cilium#33993]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

higher than v1.15.7 and lower than v1.16.0

### What happened?

I have a k8s (verison 1.30.3) cluster with 4 nodes on different public cloud platforms. The internal network IP segments are different. By using Tailscale (a VPN based on WireGuard), I have formed them into a local area network (LAN). This way, each machine has two IPs: eth0 and tailscale0. When creating the cluster, I used the Tailscale IPs. After the setup is complete, the pod networks can communicate with each other.

However, the [k8s metrics server](https://github.com/kubernetes-sigs/metrics-server/) cannot use the default release name; otherwise, the traffic will go out through eth0, making it unable to route.

```bash
helm upgrade --namespace=kube-system --install metrics-server metrics-server/metrics-server  # not ok
helm upgrade --namespace=kube-system --install metrics-1111 metrics-server/metrics-server    # ok
helm upgrade --namespace=metrics-server --create-namespace  --install metrics-server metric # ok

```

```
root@tx-sh-1:~# kubectl top node
NAME      CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%
al-sh-3   22m          2%     595Mi           32%
tx-hk-1   82m          4%     2001Mi          60%
tx-sh-1   163m         4%     3720Mi          48%
us2       13m          1%     609Mi           71%
root@tx-sh-1:~# kubectl get pod -n kube-system
NAME                                           READY   STATUS    RESTARTS   AGE
cilium-hz7wd                                   1/1     Running   0          35m
cilium-k7j2r                                   1/1     Running   0          35m
cilium-operator-6876fc4dff-2lvxn               1/1     Running   0          35m
cilium-pn57d                                   1/1     Running   0          35m
cilium-pwbw2                                   1/1     Running   0          35m
coredns-76f75df574-t4xxj                       1/1     Running   0          9h
coredns-76f75df574-zvlnw                       1/1     Running   0          22h
etcd-tx-sh-1                                   1/1     Running   34         44h
hubble-relay-6d88849768-xxbch                  1/1     Running   0          24m
hubble-ui-59bb4cb67b-xrzrp                     2/2     Running   0          24m
kube-apiserver-tx-sh-1                         1/1     Running   33         44h
kube-controller-manager-tx-sh-1                1/1     Running   9          44h
kube-scheduler-tx-sh-1                         1/1     Running   8          44h
metrics-1111-metrics-server-7d7fc9b498-n6kcq   1/1     Running   0          21m
```
```
root@tx-sh-1:~# kubectl get pod -n kube-system
NAME                               READY   STATUS             RESTARTS     AGE
cilium-hz7wd                       1/1     Running            0            44m
cilium-k7j2r                       1/1     Running            0            44m
cilium-operator-6876fc4dff-2lvxn   1/1     Running            0            44m
cilium-pn57d                       1/1     Running            0            44m
cilium-pwbw2                       1/1     Running            0            44m
coredns-76f75df574-t4xxj           1/1     Running            0            9h
coredns-76f75df574-zvlnw           1/1     Running            0            22h
etcd-tx-sh-1                       1/1     Running            34           45h
hubble-relay-6d88849768-xxbch      1/1     Running            0            33m
hubble-ui-59bb4cb67b-xrzrp         2/2     Running            0            33m
kube-apiserver-tx-sh-1             1/1     Running            33           45h
kube-controller-manager-tx-sh-1    1/1     Running            9            45h
kube-scheduler-tx-sh-1             1/1     Running            8            45h
metrics-server-7998667b79-4thp5    0/1     CrashLoopBackOff   4 (1s ago)   2m36s
root@tx-sh-1:~# kubectl top node
error: Metrics API not available
```
<details>
  <summary>kubectl logs -n kube-system metrics-server-7998667b79-4thp5</summary>

```
root@tx-sh-1:~#
Error: unable to load configmap based request-header-client-ca-file: Get "https://10.10.0.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication": dial tcp 10.10.0.1:443: i/o timeout
Usage:
   [flags]

Metrics server flags:

      --kubeconfig string            The path to the kubeconfig used to connect to the Kubernetes API server and the Kubelets (defaults to in-cluster config)
      --metric-resolution duration   The resolution at which metrics-server will retain metrics, must set value at least 10s. (default 1m0s)
      --version                      Show version

Kubelet client flags:

      --deprecated-kubelet-completely-insecure    DEPRECATED: Do not use any encryption, authorization, or authentication when communicating with the Kubelet. This is rarely the right option, since it leaves kubelet communication completely insecure.  If you encounter auth errors, make sure you've enabled token webhook auth on the Kubelet, and if you're in a test cluster with self-signed Kubelet certificates, consider using kubelet-insecure-tls instead.
      --kubelet-certificate-authority string      Path to the CA to use to validate the Kubelet's serving certificates.
      --kubelet-client-certificate string         Path to a client cert file for TLS.
      --kubelet-client-key string                 Path to a client key file for TLS.
      --kubelet-insecure-tls                      Do not verify CA of serving certificates presented by Kubelets.  For testing purposes only.
      --kubelet-port int                          The port to use to connect to Kubelets. (default 10250)
      --kubelet-preferred-address-types strings   The priority of node address types to use when determining which address to use to connect to a particular node (default [Hostname,InternalDNS,InternalIP,ExternalDNS,ExternalIP])
      --kubelet-request-timeout duration          The length of time to wait before giving up on a single request to Kubelet. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). (default 10s)
      --kubelet-use-node-status-port              Use the port in the node status. Takes precedence over --kubelet-port flag.
  -l, --node-selector string                      Selector (label query) to filter on, not including uninitialized ones, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2).

Apiserver secure serving flags:

      --bind-address ip                        The IP address on which to listen for the --secure-port port. The associated interface(s) must be reachable by the rest of the cluster, and by CLI/web clients. If blank or an unspecified address (0.0.0.0 or ::), all interfaces and IP address families will be used. (default 0.0.0.0)
      --cert-dir string                        The directory where the TLS certs are located. If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored. (default "apiserver.local.config/certificates")
      --http2-max-streams-per-connection int   The limit that the server gives to clients for the maximum number of streams in an HTTP/2 connection. Zero means to use golang's default.
      --permit-address-sharing                 If true, SO_REUSEADDR will be used when binding the port. This allows binding to wildcard IPs like 0.0.0.0 and specific IPs in parallel, and it avoids waiting for the kernel to release sockets in TIME_WAIT state. [default=false]
      --permit-port-sharing                    If true, SO_REUSEPORT will be used when binding the port, which allows more than one instance to bind on the same address and port. [default=false]
      --secure-port int                        The port on which to serve HTTPS with authentication and authorization. If 0, don't serve HTTPS at all. (default 443)
      --tls-cert-file string                   File containing the default x509 Certificate for HTTPS. (CA cert, if any, concatenated after server cert). If HTTPS serving is enabled, and --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key are generated for the public address and saved to the directory specified by --cert-dir.
      --tls-cipher-suites strings              Comma-separated list of cipher suites for the server. If omitted, the default Go cipher suites will be used.
                                               Preferred values: TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_256_GCM_SHA384.
                                               Insecure values: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_RC4_128_SHA, TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_RC4_128_SHA, TLS_RSA_WITH_3DES_EDE_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_RC4_128_SHA.
      --tls-min-version string                 Minimum TLS version supported. Possible values: VersionTLS10, VersionTLS11, VersionTLS12, VersionTLS13
      --tls-private-key-file string            File containing the default x509 private key matching --tls-cert-file.
      --tls-sni-cert-key namedCertKey          A pair of x509 certificate and private key file paths, optionally suffixed with a list of domain patterns which are fully qualified domain names, possibly with prefixed wildcard segments. The domain patterns also allow IP addresses, but IPs should only be used if the apiserver has visibility to the IP address requested by a client. If no domain patterns are provided, the names of the certificate are extracted. Non-wildcard matches trump over wildcard matches, explicit domain patterns trump over extracted names. For multiple key/certificate pairs, use the --tls-sni-cert-key multiple times. Examples: "example.crt,example.key" or "foo.crt,foo.key:*.foo.com,foo.com". (default [])

Apiserver authentication flags:

      --authentication-kubeconfig string                  kubeconfig file pointing at the 'core' kubernetes server with enough rights to create tokenreviews.authentication.k8s.io.
      --authentication-skip-lookup                        If false, the authentication-kubeconfig will be used to lookup missing authentication configuration from the cluster.
      --authentication-token-webhook-cache-ttl duration   The duration to cache responses from the webhook token authenticator. (default 10s)
      --authentication-tolerate-lookup-failure            If true, failures to look up missing authentication configuration from the cluster are not considered fatal. Note that this can result in authentication that treats all requests as anonymous.
      --client-ca-file string                             If set, any request presenting a client certificate signed by one of the authorities in the client-ca-file is authenticated with an identity corresponding to the CommonName of the client certificate.
      --requestheader-allowed-names strings               List of client certificate common names to allow to provide usernames in headers specified by --requestheader-username-headers. If empty, any client certificate validated by the authorities in --requestheader-client-ca-file is allowed.
      --requestheader-client-ca-file string               Root certificate bundle to use to verify client certificates on incoming requests before trusting usernames in headers specified by --requestheader-username-headers. WARNING: generally do not depend on authorization being already done for incoming requests.
      --requestheader-extra-headers-prefix strings        List of request header prefixes to inspect. X-Remote-Extra- is suggested. (default [x-remote-extra-])
      --requestheader-group-headers strings               List of request headers to inspect for groups. X-Remote-Group is suggested. (default [x-remote-group])
      --requestheader-username-headers strings            List of request headers to inspect for usernames. X-Remote-User is common. (default [x-remote-user])

Apiserver authorization flags:

      --authorization-always-allow-paths strings                A list of HTTP paths to skip during authorization, i.e. these are authorized without contacting the 'core' kubernetes server. (default [/healthz,/readyz,/livez])
      --authorization-kubeconfig string                         kubeconfig file pointing at the 'core' kubernetes server with enough rights to create subjectaccessreviews.authorization.k8s.io.
      --authorization-webhook-cache-authorized-ttl duration     The duration to cache 'authorized' responses from the webhook authorizer. (default 10s)
      --authorization-webhook-cache-unauthorized-ttl duration   The duration to cache 'unauthorized' responses from the webhook authorizer. (default 10s)

Apiserver audit log flags:

      --audit-log-batch-buffer-size int             The size of the buffer to store events before batching and writing. Only used in batch mode. (default 10000)
      --audit-log-batch-max-size int                The maximum size of a batch. Only used in batch mode. (default 1)
      --audit-log-batch-max-wait duration           The amount of time to wait before force writing the batch that hadn't reached the max size. Only used in batch mode.
      --audit-log-batch-throttle-burst int          Maximum number of requests sent at the same moment if ThrottleQPS was not utilized before. Only used in batch mode.
      --audit-log-batch-throttle-enable             Whether batching throttling is enabled. Only used in batch mode.
      --audit-log-batch-throttle-qps float32        Maximum average number of batches per second. Only used in batch mode.
      --audit-log-compress                          If set, the rotated log files will be compressed using gzip.
      --audit-log-format string                     Format of saved audits. "legacy" indicates 1-line text format for each event. "json" indicates structured json format. Known formats are legacy,json. (default "json")
      --audit-log-maxage int                        The maximum number of days to retain old audit log files based on the timestamp encoded in their filename.
      --audit-log-maxbackup int                     The maximum number of old audit log files to retain. Setting a value of 0 will mean there's no restriction on the number of files.
      --audit-log-maxsize int                       The maximum size in megabytes of the audit log file before it gets rotated.
      --audit-log-mode string                       Strategy for sending audit events. Blocking indicates sending events should block server responses. Batch causes the backend to buffer and write events asynchronously. Known modes are batch,blocking,blocking-strict. (default "blocking")
      --audit-log-path string                       If set, all requests coming to the apiserver will be logged to this file.  '-' means standard out.
      --audit-log-truncate-enabled                  Whether event and batch truncating is enabled.
      --audit-log-truncate-max-batch-size int       Maximum size of the batch sent to the underlying backend. Actual serialized size can be several hundreds of bytes greater. If a batch exceeds this limit, it is split into several batches of smaller size. (default 10485760)
      --audit-log-truncate-max-event-size int       Maximum size of the audit event sent to the underlying backend. If the size of an event is greater than this number, first request and response are removed, and if this doesn't reduce the size enough, event is discarded. (default 102400)
      --audit-log-version string                    API group and version used for serializing audit events written to log. (default "audit.k8s.io/v1")
      --audit-policy-file string                    Path to the file that defines the audit policy configuration.
      --audit-webhook-batch-buffer-size int         The size of the buffer to store events before batching and writing. Only used in batch mode. (default 10000)
      --audit-webhook-batch-max-size int            The maximum size of a batch. Only used in batch mode. (default 400)
      --audit-webhook-batch-max-wait duration       The amount of time to wait before force writing the batch that hadn't reached the max size. Only used in batch mode. (default 30s)
      --audit-webhook-batch-throttle-burst int      Maximum number of requests sent at the same moment if ThrottleQPS was not utilized before. Only used in batch mode. (default 15)
      --audit-webhook-batch-throttle-enable         Whether batching throttling is enabled. Only used in batch mode. (default true)
      --audit-webhook-batch-throttle-qps float32    Maximum average number of batches per second. Only used in batch mode. (default 10)
      --audit-webhook-config-file string            Path to a kubeconfig formatted file that defines the audit webhook configuration.
      --audit-webhook-initial-backoff duration      The amount of time to wait before retrying the first failed request. (default 10s)
      --audit-webhook-mode string                   Strategy for sending audit events. Blocking indicates sending events should block server responses. Batch causes the backend to buffer and write events asynchronously. Known modes are batch,blocking,blocking-strict. (default "batch")
      --audit-webhook-truncate-enabled              Whether event and batch truncating is enabled.
      --audit-webhook-truncate-max-batch-size int   Maximum size of the batch sent to the underlying backend. Actual serialized size can be several hundreds of bytes greater. If a batch exceeds this limit, it is split into several batches of smaller size. (default 10485760)
      --audit-webhook-truncate-max-event-size int   Maximum size of the audit event sent to the underlying backend. If the size of an event is greater than this number, first request and response are removed, and if this doesn't reduce the size enough, event is discarded. (default 102400)
      --audit-webhook-version string                API group and version used for serializing audit events written to webhook. (default "audit.k8s.io/v1")

Features flags:

      --contention-profiling           Enable block profiling, if profiling is enabled
      --debug-socket-path string       Use an unprotected (no authn/authz) unix-domain socket for profiling with the given path
      --enable-priority-and-fairness   If true, replace the max-in-flight handler with an enhanced one that queues and dispatches with priority and fairness (default true)
      --profiling                      Enable profiling via web interface host:port/debug/pprof/ (default true)

Logging flags:

      --log-flush-frequency duration         Maximum number of seconds between log flushes (default 5s)
      --log-json-info-buffer-size quantity   [Alpha] In JSON format with split output streams, the info messages can be buffered for a while to increase performance. The default value of zero bytes disables buffering. The size can be specified as number of bytes (512), multiples of 1000 (1K), multiples of 1024 (2Ki), or powers of those (3M, 4G, 5Mi, 6Gi). Enable the LoggingAlphaOptions feature gate to use this.
      --log-json-split-stream                [Alpha] In JSON format, write error messages to stderr and info messages to stdout. The default is to write a single stream to stdout. Enable the LoggingAlphaOptions feature gate to use this.
      --logging-format string                Sets the log format. Permitted formats: "json" (gated by LoggingBetaOptions), "text". (default "text")
  -v, --v Level                              number for the log level verbosity
      --vmodule pattern=N,...                comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)

panic: unable to load configmap based request-header-client-ca-file: Get "https://10.10.0.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication": dial tcp 10.10.0.1:443: i/o timeout

goroutine 1 [running]:
main.main()
        sigs.k8s.io/metrics-server/cmd/metrics-server/metrics-server.go:37 +0x8b
```
</details>

### How can we reproduce the issue?

Install metrics server with default name
```shell
helm upgrade --namespace=kube-system --install metrics-server metrics-server/metrics-server  # not ok
```

### Cilium Version

```
cilium version
cilium-cli: v0.16.13 compiled with go1.22.5 on linux/amd64
cilium image (default): v1.15.6
cilium image (stable): v1.15.7
cilium image (running): 1.15.7
```

```
helm upgrade --install cilium cilium/cilium --version 1.15.7 \
    --namespace kube-system \
    --set kubeProxyReplacement=true \
    --set k8sServiceHost=100.99.240.56 \
    --set ipam.operator.clusterPoolIPv4PodCIDRList=10.11.0.0/16 \
    --set k8sServicePort=6443 \
    --set hubble.relay.enabled=true \
    --set hubble.ui.enabled=true \
    --set operator.replicas=1 \
    --set bpf.masquerade=true \
    --set bpf.lbExternalClusterIP=true
```

### Kernel Version

```
uname -a
Linux tx-sh-1 6.1.0-23-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.99-1 (2024-07-15) x86_64 GNU/Linux
```

### Kubernetes Version

```shell
kubectl version
Client Version: v1.30.3
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.3
```

```
root@tx-sh-1:~# kubectl get node -owide
NAME      STATUS                     ROLES           AGE    VERSION   INTERNAL-IP       EXTERNAL-IP   OS-IMAGE                         KERNEL-VERSION     CONTAINER-RUNTIME
al-sh-3   Ready                      <none>          45h    v1.30.3   100.101.129.114   <none>        Ubuntu 22.04.4 LTS               6.5.0-44-generic   containerd://1.7.19
tx-hk-1   Ready,SchedulingDisabled   <none>          31h    v1.30.3   100.123.67.62     <none>        Ubuntu 22.04.4 LTS               6.5.0-44-generic   containerd://1.7.19
tx-sh-1   Ready                      control-plane   260d   v1.30.3   100.99.240.56     <none>        Debian GNU/Linux 12 (bookworm)   6.1.0-23-amd64     containerd://1.7.19
us2       Ready,SchedulingDisabled   <none>          30h    v1.30.3   100.111.209.82    <none>        Ubuntu 22.04.4 LTS               6.5.0-44-generic   containerd://1.7.19
```

### Sysdump

[cilium-sysdump-20240724-222842.zip](https://github.com/user-attachments/files/16363596/cilium-sysdump-20240724-222842.zip)

### Anything else?

If I add a route to main table, it work well.
![image](https://github.com/user-attachments/assets/01d6dcd4-542c-4eb7-9c3d-f7b827c777fc)

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by squeed** [comment]
I'm a bit confused. Why does the release name of the Helm chart make a difference? What, exactly, is not working with the metrics server? Can you post the failure?

**Comment by qianlongzt** [comment]
After upgrading to 1.16.0, reinstalling 1.15.7 works without any issues. I can't reproduce it. Just close it
